File: src/test/java/org/snake/models/basic/VectorTest.java
package org.snake.models.basic;

import org.junit.Test;

import javax.xml.bind.util.ValidationEventCollector;

import static org.junit.Assert.*;

public class VectorTest {

    @Test
    public void isOrigin() {
        assertTrue(new Vector(0, 0).isOrigin());
        assertFalse(new Vector(1, 1).isOrigin());
        assertFalse(new Vector(-11, -1).isOrigin());
    }

    @Test
    public void addPoint() {
        // Arrange
        Vector vector = new Vector(1, 1);

        // Act
        Point point = vector.add(new Point(1, 1));

        // Assert
        assertEquals(new Point(2, 2), point);
    }

    @Test
    public void addVector() {
        // Arrange
        Vector vector = new Vector(1, 1);

        // Act
        Vector result = vector.add(new Vector(1, 1));

        // Assert
        assertEquals(new Vector(2, 2), result);
    }
}

File: src/test/kotlin/org/snake/models/basic/VectorTest.kt
package org.snake.models.basic

import org.junit.Assert
import org.junit.Test
import snake.models.basic.Point
import snake.models.basic.Vector

class VectorTest {

    @Test
    fun isOrigin(): Unit {
        Assert.assertTrue(Vector(0, 0).isOrigin)
        Assert.assertFalse(Vector(1, 1).isOrigin)
        Assert.assertFalse(Vector(-11, -1).isOrigin)
    }

    @Test
    fun addPoint() {
        // Arrange
        val vector = Vector(1, 1)

        // Act
        val point = vector.add(Point(1, 1))

        // Assert
        Assert.assertEquals(Point(2, 2), point)
    }

    @Test
    fun addVector() {
        // Arrange
        val vector = Vector(1, 1)

        // Act
        val result = vector.add(Vector(1, 1))

        // Assert
        Assert.assertEquals(Vector(2, 2), result)
    }
}